onnxruntime/core/providers/coreml/builders/impl/builder_utils.h

```diff
@@ -136,6 +136,17 @@ void AddOperationInput(COREML_SPEC::MILSpec::Operation& op,
 /// <param name="output">NodeArg with details of output to add.</param>
 void AddOperationOutput(COREML_SPEC::MILSpec::Operation& op, const NodeArg& output);
 
+/// <summary>
+/// Add an output to a MILSpec::Operation for an intermediate operation when the implementation is composed of
+/// multiple MLProgram operations. In this case we don't have a NodeArg for the output.
+/// </summary>
+/// <param name="op">Operation to update.</param>
+/// <param name="name">Name of the intermediate output.</param>
+/// <param name="element_type">onnx::TensorProto_DataType element type of the output.
+///   int32_t as that is what TensorShapeProto uses to store the value.</param>
+/// <param name="shape">Shape of the output if known.</param>
+void AddIntermediateOperationOutput(COREML_SPEC::MILSpec::Operation& op, const std::string& name,
```

Comment:
use std::string_view for consistency with other functions like AddOperationInput()?

Reply:
Using as the input should always be coming from ModelBuilder::GetUniqueName which returns a std::string, and we can avoid a copy when calling `set_name` this way. For AddOperationInput we're usually taking const char* input.